src/main/java/org/terasology/signalling/componentSystem/SignalNetworkNode.java

```diff
@@ -26,17 +26,24 @@
         PRODUCER, CONSUMER, CONDUCTOR
     }
 
+    // The node type
     private Type type;
 
     public SignalNetworkNode(Vector3i location, byte inputSides, byte outputSides, Type type) {
         super(location, inputSides, outputSides);
         this.type = type;
     }
 
+    /**
+     * @return The type of node this node is. Can be any of {@link SignalNetworkNode.Type}
+     */
     public Type getType() {
         return type;
     }
 
+    /**
+     * @return True if the given object is the same node or a node of the same {@link SignalNetworkNode.Type}
```

Comment:
This function does not need javadocs

Reply:
If anything, add a single line to the javadoc of the class.
`Note that two nodes will be considered same if they have the same Type` or something.